src/P3Scheme.jl

```diff
@@ -667,4 +767,464 @@ function D_m(p3::PSP3, q::FT, N::FT, ρ_r::FT, F_r::FT) where {FT}
     return n / q
 end
 
+"""
+    get_rain_parameters(q, N) 
+
+ - q - mass mixing ratio of rain
+ - N - number mixing ratio of rain 
+
+ returns the parameters λ and N_0 where N' = N_0 * exp(-λ  * D)
+"""
+function get_rain_parameters(q::FT, N::FT) where {FT}
+    ρ_w = FT(1000)
+    λ = (π * ρ_w * N / q)^(1 / 3)
+    N_0 = N * λ
+    return (λ, N_0)
+end
+
+"""
+    get_cloud_parameters(q, N) 
+
+ - q - mass mixing ratio of rain
+ - N - number mixing ratio of rain 
+
+ returns the parameters λ and N_0 where N' = N_0 * D ^ 8 * exp(-λ ^3  * D ^ 3)
+"""
+function get_cloud_parameters(q::FT, N::FT) where {FT}
+    ρ_w = FT(1000)
+    ρ_a = FT(1.2)
+    λ = (π * ρ_w * N / (2 * q * ρ_a))^(1 / 3)
+    N_0 = 3 / 2 * N * λ^9
+    return (λ, N_0)
+end
+
+"""
+    N′_rain(D, q, N)
+
+ - D - diameter of particle 
+ - q - mass mixing ratio
+ - N - number mixing ratio 
+
+ Returns the distribution of rain particles (assumed to be of the form 
+ N'(D) = N0 * exp(-λD)) at given D 
+"""
+function N′rain(D::FT, q::FT, N::FT) where {FT}
+    (λ, N_0) = get_rain_parameters(q, N)
+    return N_0 * exp(-λ * D)
+end
+
+"""
+    N′_cloud(D, q, N)
+
+ - D - diameter of particle 
+ - q - mass mixing ratio
+ - N - number mixing ratio 
+
+ Returns the distribution of cloud particles (assumed to be of the form 
+ N'(D) = N0 * D ^ 8 * exp(-λ D ^ 3)) at given D 
+"""
+function N′cloud(D::FT, q::FT, N::FT) where {FT}
+    (λ, N_0) = get_cloud_parameters(q, N)
+    return N_0 * D^8 * exp(-λ^3 * D^3)
+end
+
+"""
+    N′_ice(D, p3, λ, N0)
+
+ - D - diameter of particle 
+ - p3 - a struct containing P3 scheme parameters
+ - λ - shape parameter of distribution 
+ - N0 - shape parameter of distribution
+
+ Returns the distribution of ice particles (assumed to be of the form 
+ N'(D) = N0 * D ^ μ * exp(-λD)) at given D 
+"""
+function N′ice(D::FT, p3::PSP3, λ::FT, N_0::FT) where {FT}
+    return N_0 * D^DSD_μ(p3, λ) * exp(-λ * D)
+end
+
+"""
+    ϕ_ice(p3, D, F_r, th) 
+
+ - p3 - a struct containing P3 Scheme parameters 
+ - D - maximum particle dimension
+ - F_r - rime mass fraction (q_rim/q_i)
+ - th - P3Scheme nonlinear solve output tuple (D_cr, D_gr, ρ_g, ρ_d)
+
+ Returns the ϕ calculation for an ice particle in the P3 Scheme with given 
```

Comment:
What is `ϕ` ?

Reply:
Should this maybe be moved to the same place where we calculate terminal velocities and have a passed in mass and area? This is the only place where phi is not 1 but for future use maybe it would be more helpful there?